datajunction-server/datajunction_server/construction/build.py

```diff
@@ -702,6 +712,47 @@ def build_node(  # pylint: disable=too-many-arguments
     return built_ast
 
 
+def rename_columns(built_ast: ast.Query, node: NodeRevision):
+    """
+    Rename columns in the built ast to fully qualified column names.
+    """
+    projection = []
+    node_columns = {col.name for col in node.columns}
+    for expression in built_ast.select.projection:
+        if (
+            not isinstance(expression, ast.Alias)
+            and not isinstance(
+                expression,
+                ast.Wildcard,
+            )
+            and not (hasattr(expression, "alias") and expression.alias)  # type: ignore
+        ):
+            alias_name = expression.alias_or_name.identifier(False)  # type: ignore
+            if expression.alias_or_name.name in node_columns:  # type: ignore
+                alias_name = node.name + SEPARATOR + expression.alias_or_name.name  # type: ignore
+            expression = expression.copy()
+            expression.set_alias(ast.Name(amenable_name(alias_name)))
+            projection.append(expression)
+        else:
+            expression = expression.copy()
+            projection.append(expression)  # type: ignore
+    built_ast.select.projection = projection
+
+    if built_ast.select.where:
+        for col in built_ast.select.where.find_all(ast.Column):
+            if hasattr(col, "alias"):  # pragma: no cover
+                col.alias = None
```

Comment:
Why do we need to cleanup the aliases in WHERE and GROUP BY?

Reply:
@agorajek https://github.com/DataJunction/dj/blob/347abfd961bab9aab3f273aedcbe262cd2e24e16/datajunction-server/datajunction_server/sql/parsing/ast.py#L1005 aliases get appended when SQL is rendered. There's no check on the ancestor nodes to determine if the alias would be valid.